userProfile/src/Components/RenderForm.tsx

```diff
@@ -0,0 +1,59 @@
+import { Button, Gapped, Input, Select } from '@skbkontur/react-ui';
+import React, { useState } from 'react';
+import { FormData, defaultData } from './Form';
+
+type TFormProps = {
+    data: FormData;
+    onSubmit: Function;
+};
+
+const cities: string[] = ['Екатеринбург', 'Н.Тагил', 'Москва', 'Санкт-Петербург', 'Казань'];
+
+const RenderModal = (props: TFormProps) => {
+    const [person, setPerson] = useState({ ...defaultData });

```

Comment:
Вероятно она победила дважды, и тут должно было быть
```suggestion
    const [person, setPerson] = useState({ ...props.data });
```
А то `data` вроде передаётся, но нигде не используется

Reply:
На самом деле этому компоненту не нужен стейт, он может работать со стейтом копонента `Form`.
Если объявить `handlerField` в `Form`, а из функции `updatePerson` выкинуть `setNewPerson({ ...newPerson, ...data });` и передавать в `RenderForm` пропсы `person`, `handlerField` и `closeModal`, то всё продолжит работать :)